### PR TITLE
Add RFC document structure template

### PR DESCRIPTION
**Pull request by mmmarcos**
This PR adds a document structure template to write RFC for Parsec. The goal is to standardize these technical documents to make them easier to read and to find important information.

This is inspired by https://wasp-lang.dev/blog/2023/12/05/writing-rfcs.

### CODE REVIEW

File: docs/rfcs/0000-rfc-template.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+## Overview
+
+> If someone only reads this, what do you want them to know?
+
+## Goals and Non-Goals
+
+> What problems are you trying to solve?
+> What problems are you not trying to solve? (this can help better define scope).
+>
```
Comment (FirelightFlagboy) [inline docs/rfcs/0000-rfc-template.md:15]:
```suggestion

```

File: docs/rfcs/0000-rfc-template.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Overview
+
+> If someone only reads this, what do you want them to know?
+
+## Goals and Non-Goals
+
+> What problems are you trying to solve?
+> What problems are you not trying to solve? (this can help better define scope).
+## Background & Motivation
```
Comment (mmmarcos) [inline docs/rfcs/0000-rfc-template.md:15]:
```suggestion

## Background & Motivation
```

File: docs/rfcs/0000-rfc-template.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+# [RFC] Title here
+
+> Delete lines starting with ">". Remove sections that are not applicable.
+
+## Overview
+
+> If someone only reads this, what do you want them to know?
+
+## Goals and Non-Goals
```
Comment (FirelightFlagboy) [inline docs/rfcs/0000-rfc-template.md:11]: I would recommend having `Background & Motivation` before that point.

IMO it seems more natural :shrug:

File: docs/rfcs/0000-rfc-template.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+> What problems are you trying to solve?
```
Comment (FirelightFlagboy) [inline docs/rfcs/0000-rfc-template.md:13]: This question is pretty similar to `> Why is this change being proposed?` (in Background Motivation)

File: docs/rfcs/0000-rfc-template.md (hunk 3 above)
Comment (mmmarcos) [reply docs/rfcs/0000-rfc-template.md]: Done

File: docs/rfcs/0000-rfc-template.md (hunk 4 above)
Comment (mmmarcos) [reply docs/rfcs/0000-rfc-template.md]: Done

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**